snuba/datasets/discover.py

```diff
@@ -65,52 +132,19 @@ def detect_table(
     Given a query, we attempt to guess whether it is better to fetch data from the
     "events" or "transactions" storage. This is going to be wrong in some cases.
     """
-    event_columns = set()
-    transaction_columns = set()
-    for col in query.get_all_referenced_columns():
-        if events_only_columns.get(col):
-            event_columns.add(col)
-        elif transactions_only_columns.get(col):
-            transaction_columns.add(col)
-    for col in query.get_all_ast_referenced_columns():
-        if events_only_columns.get(col.column_name):
-            event_columns.add(col.column_name)
-        elif transactions_only_columns.get(col.column_name):
-            transaction_columns.add(col.column_name)
-
-    selected_table = None
-
-    # First check for a top level condition that matches either type = transaction
-    # type != transaction.
-    conditions = query.get_conditions()
-    if conditions:
-        for idx, condition in enumerate(conditions):
-            if is_condition(condition):
-                if tuple(condition) == ("type", "=", "error"):
-                    selected_table = EVENTS
-                elif tuple(condition) == ("type", "=", "transaction"):
-                    selected_table = TRANSACTIONS
```

Comment:
Did this actually work as expected?
A condition between the column type and a literal should have been
`("type", "=", "'error'")` and not `("type", "=", "error")`

Or did we just send a wrong query in order to fit this condition ?

Reply:
oh damn, I just noticed why this works, which is horrendous. The old language parser assumes everything on the right hand side of a condition to be a literal so it does not require quotes ..... this is horrible.

Reply:
This check, while functional, was always horrible. I'm glad to be using the AST instead.